pallets/schemas/src/lib.rs

```diff
@@ -316,11 +316,16 @@ impl<T: Config> SchemaProvider<SchemaId> for Pallet<T> {
 	fn get_schema_by_id(schema_id: SchemaId) -> Option<SchemaResponse> {
 		// To account for db read
 		Self::get_schema_by_id(schema_id);
+
+		// This is janky, but it allows us to benchmark both message extrinsics from
+		// this pallet.
+		let location: PayloadLocation =
```

Comment:
is it possible to define a method on `PayloadLocation` with `#[cfg(feature = "runtime-benchmarks")]` that gives you location by schema_id and you move this logic inside that.

Reply:
we'd have to define this method on the `SchemaProvider` class. If we're ok with that (as in, defining a method purely for the purpose of benchmarking) then we can do it.